File: Data Structures & Algorithms/is-anagram/submission-1.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        a={}
        for i in s:
            if i not in a:
                a[i]=1
            else:
                a[i]+=1

        for i in t:
            if i not in a:
                return False
            else:
                a[i]-=1
        
        for i in a:
            if a[i]!=0:
                return False
        return True

            


            
        

File: Data Structures & Algorithms/anagram-groups/submission-1.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string,vector<string>> res;
        for(string s:strs){
            string sorteds=s;
            sort(sorteds.begin(),sorteds.end());
            res[sorteds].push_back(s);
        }
        vector<vector<string>> result;
        for (auto& x:res){
            result.push_back(x.second);
        }
        return result;
    }
};
